### ISSUE TITLE
No audio AC-3 over HDMI on Colossus2 with HDMI only connection

### ISSUE THREAD

**Issue by jhoyt4** [jpoet/HauppaugeUSB#21]
I am unable to record audio using the HDMI with audio codec set to AC-3.

AAC audio over HDMI works, AC-3 over SPIDF works.

I am using a colossus 2 that has SPIDF input.

/opt/Hauppauge/bin/hauppauge2 -s E585-00-00D642B7 -a 3 -d 2 -i 3 -o /tmp/temp.ts --v   --loglevel debug

output attached
[output.txt](https://github.com/jpoet/HauppaugeUSB/files/4706266/output.txt)

**Comment by jhoyt4** [comment]
...corrected the title.  Apologies, I just noticed I called out the wrong problem.

**Comment by jpoet** [comment]
Just to verify, you have built the latest code from this repository, from scratch, including applying the latest patches to the Hauppauge source?  I ask because I ran into an issue lately, where I didn't think to apply the "new" patches against the Hauppauge code.

**Comment by jhoyt4** [comment]
I was running off of version built May 20 (commit 317c249) from my launchpad ppa.

I just pushed a "clean" version where I pulled everything fresh down from github and pushed to my ppa.  Unfortunately this version did not work (e.g. no audio for AC-3 over HDMI).

I will try rebuilding per the github instruction (e.g. no ppa nonsense) and see if that fixes the issue.

**Comment by jhoyt4** [comment]
Confirmed - still no audio on a completely fresh build (no ppa).

Thanks for looking into this.

[output.noppa.txt](https://github.com/jpoet/HauppaugeUSB/files/4738342/output.noppa.txt)

**Comment by jpoet** [comment]
The only thing I can think of, is that your "STB" is not configured to send AC-3 via HDMI. With that latest code, the Hauppauge should be advertising that it can accept AC-3 via its EDID, but if the STB has that disabled for some reason, it wont do it.

**Comment by jhoyt4** [comment]
I just looked into the STB's advanced menu and for some reason HDMI is locked at 2CH output.  This looks to be a my configuration problem.

Apologies for the distraction and thank you for the help!!!

**Comment by jhoyt4** [comment]
I just confirmed AC-3 over HDMI does work very well.

My issue was absolutely a HW configuration error.  An HDMI extender I was using was stripping the audio down to 2-channel AAC despite being advertised as full HDMI 1.3 compliant with 5.1 channel audio.  I replaced the extender and everything worked for AC-3.

Apologies again for stealing your time and thank you very much for the pointer that helped reveal the real solution.
